selfdrive/navd/navd.py

```diff
@@ -327,7 +338,33 @@ def reset_recompute_limits(self):
     self.recompute_backoff = 0
     self.recompute_countdown = 0
 
-  def should_recompute(self):
+  def should_recompute(self, new_destination: Optional[Coordinate]) -> Tuple[bool, Optional[RECOMPUTE_REASON]]:
+    # Don't recompute when GPS drifts in tunnels
+    if not self.gps_ok and self.step_idx is not None:
+      return False, None
+
+    if new_destination != self.nav_destination:
+      cloudlog.warning(f"Got new destination from NavDestination param {new_destination}")
+      return True, RECOMPUTE_REASON.NEW_DESTINATION
+
+    if self.too_far_from_route():
+      return True, RECOMPUTE_REASON.DISTANCE_TO_ROUTE
+
+    if self.going_wrong_direction():
+      return True, RECOMPUTE_REASON.WRONG_DIRECTION
+
+    return False, None
```

Comment:
Is this required for this PR?

Reply:
not required at all, was just for the suggestion of logging the reason for recompute. I'll work on pulling it out